server/src/main/java/org/candlepin/resteasy/DateFormatter.java

```diff
@@ -15,34 +15,74 @@
 package org.candlepin.resteasy;
 
 import org.apache.commons.lang3.StringUtils;
+
 import org.jboss.resteasy.spi.StringParameterUnmarshaller;
 import org.jboss.resteasy.util.FindAnnotation;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 import java.lang.annotation.Annotation;
 import java.text.ParseException;
 import java.text.SimpleDateFormat;
 import java.util.Date;
 
+import javax.xml.bind.DatatypeConverter;
+
+
+
 /**
- * Formatter for dates on REST API parameters
+ * Formatter for parsing dates for parameters annotated with the DateFormat annotation.
  */
 public class DateFormatter implements StringParameterUnmarshaller<Date> {
-    private SimpleDateFormat formatter;
+    private static Logger log = LoggerFactory.getLogger(DateFormatter.class);
+
+    private String[] formats;
 
     public void setAnnotations(Annotation[] annotations) {
-        DateFormat format = FindAnnotation.findAnnotation(annotations, DateFormat.class);
-        formatter = new SimpleDateFormat(format.value());
+        DateFormat annotation = FindAnnotation.findAnnotation(annotations, DateFormat.class);
+
+        this.formats = annotation.value();
     }
 
-    public Date fromString(String str) {
-        if (StringUtils.isBlank(str)) {
-            return new Date(0);
-        }
-        try {
-            return formatter.parse(str);
-        }
-        catch (ParseException e) {
-            throw new RuntimeException(e);
+    public Date fromString(String value) {
+        // If we're null/empty, we have nothing to parse, so return null
+        if (!StringUtils.isBlank(value)) {
+            // If the value is our special value for "now", return a new Date instance
+            if (DateFormat.NOW.equals(value)) {
+                return new Date();
+            }
+
+            // If we have any formats specified, use those...
+            if (this.formats != null && this.formats.length > 0) {
+                for (String format : this.formats) {
+                    try {
+                        SimpleDateFormat formatter = new SimpleDateFormat(format);
+                        return formatter.parse(value);
+                    }
+                    catch (ParseException exception) {
+                        // Whoops. Hopefully we have more formats to try...
+                    }
+                }
+
+                // If we made it here, we're out of formats to use to parse the date
+
+                // TODO: How do we get an i18n instance at this point to translate this?
+                String message = "Unable to parse date parameter: " + value;
+                throw new RuntimeException(message);
+            }
+            else {
+                try {
```

Comment:
How about a compromise between what wottop suggested and Ceiu brought up?  I.e. the ISO8601 format is provided as a constant within the `DateFormat` annotation so that it's easy to add.  `@DateFormat(DateFormat.ISO8601)`